flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/source/lookup/cache/DefaultLookupCache.java

```diff
@@ -222,11 +248,23 @@ public Builder cacheMissingKey(boolean cacheMissingKey) {
 
         /** Creates the cache. */
         public DefaultLookupCache build() {
+            sanityCheck();
             return new DefaultLookupCache(
                     expireAfterAccessDuration,
                     expireAfterWriteDuration,
                     maximumSize,
                     cacheMissingKey);
         }
+
+        private void sanityCheck() {
+            if (expireAfterWriteDuration == null
+                    && expireAfterAccessDuration == null
+                    && maximumSize == null) {
+                throw new IllegalArgumentException(
```

Comment:
Add check for `org.apache.flink.table.connector.source.lookup.cache.DefaultLookupCache#fromConfig` as well?